app/src/main/java/skills/future/planer/db/task/TaskDataViewModel.java

```diff
@@ -34,11 +35,48 @@ public class TaskDataViewModel extends AndroidViewModel {
     public TaskDataViewModel(Application application) {
         super(application);
         mRepository = new TaskDataRepository(application);
-        allTaskData = mRepository.getListLiveData();
+        allTaskData = mRepository.getAllTaskData();
+        calendarDate = Calendar.getInstance();
+    }
+
+    public LiveData<List<TaskData>> getAllTaskData() {
+        return allTaskData;
+    }
+
+    /**
+     * Delegates getting tasks to TaskDataRepository
+     *
+     * @param quarter quarter number
+     * @param date    date for which we want to get data
+     */
+    public LiveData<List<TaskData>> getCategorizedTaskDataFromDay(int quarter, long date)
+            throws Exception {
+        return mRepository.getCategorizedListLiveDataFromDay(quarter, date);
+    }
+
+    public LiveData<List<TaskData>> getAllTaskDataFromDay(CalendarDay date) {
+        return mRepository.getAllTaskDataFromDay(convertCalendarDayToLong(date));
+    }
+
+    /**
+     * Delegates getting number of tasks to TaskDataRepository
+     *
+     * @param date date for which we want to get data
+     */
+    public int getNumberOfTaskByDate(CalendarDay date) {
+        return mRepository.getNumberOfTaskByDate(convertCalendarDayToLong(date));
+    }
+
+    /**
+     * Converts date to long value
+     */
+    private long convertCalendarDayToLong(CalendarDay date) {
+        calendarDate.set(date.getYear(), date.getMonth(), date.getDay());
+        return calendarDate.getTimeInMillis();
     }
```

Comment:
To może być metoda statyczna, wtedy będzie można ją przenieść gdzieś indziej

Reply:
Nie wiem czy będzie przenoszona, ja tego obecnie nie zakładam, więc jest prywatna